pkgs/tools/networking/pihole-ftl/default.nix

```diff
@@ -0,0 +1,50 @@
+{ cmake
+, fetchFromGitHub
+, gmp
+, lib
+, libidn
+, nettle
+, readline
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pihole-ftl";
+  version = "5.7";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "FTL";
+    rev = "v${version}";
+    sha256 = "1xls9s4ss1dxn1n5n17amanqqwnsqd5x7idkwr06z1jxnx2rf2b6";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ gmp libidn nettle readline ];
+
+  postPatch = ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX"
+
+    substituteInPlace src/version.h.in \
+      --replace "@GIT_VERSION@" "v${version}" \
+      --replace "@GIT_DATE@" "2020-12-26" \
```

Comment:
Maybe the `GIT_DATE` should be `1970-01-01`?

Reply:
I think the intention with `GIT_DATE` is to be the date the commit was tagged. But setting it to `1970-01-01` does simplify maintenance.

Reply:
Sounds about right. But I'd rather not have to change it every time it gets updated. Could also just leave it un-substituted. If the release already had this substituted, that would have been cool, but oh well.

Reply:
Unfortunately it is unset by default, and the build will fail if it is not set.